server/plugin.go

```diff
@@ -99,9 +97,6 @@ func (p *Plugin) OnActivate() error {
 	}
 	// app store
 	appstore := p.store.App
-	if pluginapi.IsConfiguredForDevelopment(mmconf) {
-		appstore.InitBuiltin(builtin_hello.App())
-	}
```

Comment:
Why isn't this needed any longer?

Reply:
@hanzei The built-in app does not have a bot, and so the features that require a bot (submitting any form) do not work, so it is sort of useless to have there at its current state. The function that creates the bot (below) requires a user session token, which is not available upon plugin startup, since there is no user interaction going on.

https://github.com/mattermost/mattermost-plugin-apps/blob/02428ac9dd1423231989d1d56fdecfe7d43da1bf/server/proxy/install.go#L139

There were a few options to choose from, in order to write an e2e test that performs some app's action. Hosting the test app or the `helloworld` app that runs in a separate port/process was not an option. Here some some options:

- Have the built-in app create its own bot on init. Since there is no user session token, we would be doing so outside of the `proxy.ensureBot` function. In doing so, we would be duplicating code and risking having these pieces of code being out of sync.
- Use the http-hello app, and keep the builtin app installed. This has the issue of the bindings' HTML ids colliding in the webapp. So the e2e test cannot objectively find the right element.
  - One solution to this would be append the app id to the binding locations when serving up the bindings, but this makes the command names change from `send` to `send-http-hello`. I chose not to do this for this reason.
  - Another solution would be to update the webapp to include the app id in the HTML id, as `${app_id}_${location}`. I'm submitting a PR to include this as this should be done anyway.
- Remove the built-in app, and only use the http-hello app in the e2e test. This has the drawback of not having a default app installed. I ultimately chose this option.

There are probably other practical options, but this is what I considered when writing the test

Reply:
The lack of a bot account for the built-in app seems like a bug to me. I was expecting the app to have them. `actingUserID` is only used for notification, which is not needed in this case. `ensureBot` could be rewritten to optionally accept a userID.